### PR TITLE
Fix hostname plugin for RedHat systems

### PR DESCRIPTION
**Pull request by florisvanstal**
Wrote as fix for ticket DIS-2796, where the hostname could not be found for Red Hat systems. The hostname for these files is instead of /etc/hostname found within /etc/sysconfig/network.

### CODE REVIEW

File: dissect/target/plugins/os/unix/_os.py [hunk 1]
```diff
@@ -139,21 +139,29 @@ def _parse_hostname_string(self, paths: Optional[list[str]] = None) -> Optional[
         Args:
             paths (list): list of paths
         """
-        paths = paths or ["/etc/hostname", "/etc/HOSTNAME"]
+        paths = paths or ["/etc/hostname", "/etc/HOSTNAME", "/etc/sysconfig/network"]
```
Comment (cecinestpasunepipe) [inline dissect/target/plugins/os/unix/_os.py:142]:
```suggestion
        REDHAT_LEGACY_PATH = "/etc/sysconfig/network"
        paths = paths or ["/etc/hostname", "/etc/HOSTNAME", REDHAT_LEGACY_PATH]
```

File: dissect/target/plugins/os/unix/_os.py [hunk 2]
```diff
@@ -139,21 +139,29 @@ def _parse_hostname_string(self, paths: Optional[list[str]] = None) -> Optional[
...
         hostname_string = None
 
         for path in paths:
             for fs in self.target.filesystems:
-                if fs.exists(path):
+                if not fs.exists(path):
+                    continue
+                if path == "/etc/sysconfig/network":
```
Comment (cecinestpasunepipe) [inline dissect/target/plugins/os/unix/_os.py:149]:
```suggestion
                if path == REDHAT_LEGACY_PATH:
```

File: dissect/target/plugins/os/unix/_os.py [hunk 3]
```diff
@@ -139,21 +139,29 @@ def _parse_hostname_string(self, paths: Optional[list[str]] = None) -> Optional[
...
-            else:
-                hostname_dict = {"hostname": hostname_string, "domain": None}
+        if hostname_string and "." in hostname_string:
+            hostname_string = hostname_string.split(".", maxsplit=1)
+            hostname_dict = {"hostname": hostname_string[0], "domain": hostname_string[1]}
+        else:
+            hostname_dict = {"hostname": hostname_string, "domain": None}
 
-            return hostname_dict
+        return hostname_dict
```
Comment (cecinestpasunepipe) [inline dissect/target/plugins/os/unix/_os.py:165]:
```suggestion
def _parse_rh_legacy(self, path):
    file_contents = fs.path(path).open("rt").readlines()
    for line in file_contents:
        if not line.startswith("HOSTNAME"):
            continue
        _,_,hostname = line.rstrip().partition("=")
        return hostname
```


Also can you add a unit test for this new scenario?

**Review by cecinestpasunepipe — CHANGES_REQUESTED** [review]

**Review by pyrco — CHANGES_REQUESTED** [review]
Could you change the commit/PR message according to https://docs.dissect.tools/en/latest/contributing/style-guide.html#commit-message-style-and-formatting

File: tests/plugins/os/unix/test__os.py [hunk 4]
```diff
@@ -61,3 +67,44 @@ def test_parse_fstab(tmp_path):
...
     }
+
+
+@pytest.mark.parametrize(
+    "path, is_domain_joined, expected_hostname, expected_domain",
+    [
+        ("/etc/hostname", True, "myhost", "mydomain.com"),
+        ("/etc/HOSTNAME", True, "myhost", "mydomain.com"),
+        ("/etc/sysconfig/network", True, "myhost", "mydomain.com"),
+        ("/etc/hostname", False, "myhost", None),
```
Comment (cecinestpasunepipe) [inline tests/plugins/os/unix/test__os.py:78]: Also add some invalid cases please:

- invalid filename
- invalid file contents
- both with domain

**Review by cecinestpasunepipe — CHANGES_REQUESTED** [review]

**Review by cecinestpasunepipe — APPROVED** [review]
Tested on RH 7, works.

**Review by cecinestpasunepipe — APPROVED** [review]

**Review by pyrco — APPROVED** [review]

**Pull request merged**